Prozess.txt

```diff
@@ -0,0 +1,240 @@
+			-einen eigenen Stapel (Stack)
+				(Stack: Ein Stapel mit Last in first Out- Prinzip)
+			
+		Registersatz:
+			Die Menge der Register im Mikroprozessor, die durch Programme direkt
+			angesprochen werden können.
+			3 Typen:
+				-Datenregister zur Zwischenpeicherung von Operanden und 
+				 Rechenergebnissen
+				-Adressregister zur Adressirung von Operanden und Befehlen.
+				-Spezialregister für unterschiedliche Anwendungen z.B Programmzähler
+				 oder Stackpointer.
+
+	Typen:
+		-User-Threads:
+			Nicht im Kernel implementiert, sondern in einer separaten
+			Programmbibliothek.
+				-Taskswitching zwischen Userthreads ohne aufwenidge Systemaufrufe
+				 möglich.
+				-Wen ein Task blcokiert sind alle blockiert.
+
+		-Kernel-Threads:
+			Im kernel implementiert
+				-Taskswichting needs Hardware Support.
+				-Wen ein Task blockiert, kann ein anderer übernehmen.
+				
+Init-Prozess:
+	Ist bei UNIX-Systemen der allererste Prozess mit Prozess-ID 1 (PID1)
+	dieser startet alle anderen Prozesse. Welche das sind und in welcher reihenfolge diese gestartet werden,
+	ist in etc/inittab festgelegt.
+	ein kill init kommt dem Herunterfahren des Systems gleich, da mit dessen Beendigung alle anderen Prozesse beendet werden.
+	Diese anderen Prozesse werden beendet da sie Kind-Prozesse des Init-Prozesses sind.
+	Als aller erstes führt der Init-Prozess jedoch einen POST (Power-On-Self-Test) durch,
+	bei diesem wird:
+		-Funktion der CPU
+		-Funktion der RAM
+		-Funktion der Cache
+		-Funktion der I/O-Software
+	geprüft.
+
+Shell:
+	Software die das Interagieren mit dem OS möglich macht. 
+	Kernel = Kern des OS, Shell = Hülle, für den benutzer Sichtbarer teil des OS.
+
+
+Standart-streams:
+	Standart-Datenströme sind 3 Datenströme für die Ein- und Ausgabe im Betriebsystem.
+		-stdin: Input vom z.B. Keybord
+		-stdout: Ausgabe von z.B Text auf dem Display
+		-stderr: Ausgabe von Fehlermeldungen auf dem Display oder Ton.
+
+	Standarteingaben:
+		Über Standarteingaben können Daten in ein Programm eingelesen werden.
+			-Standarteingabe die Gerätedatei /dev/stdin, Deskriptor Nummer 0
+			-Bsp. $less < eingaben.txt:
+				-Standarteingabe für das Programm less aus der Datei eingaben.txt
+
+	Standartausgaben:
+		Über Standartausgaben kann ein Programm Daten ausgeben.
+			Standartausgaben die Gerätedatei /dev/stdout, Deskriptor Nummer 1
+			Bsp. $find . -name "*html" > ausgaben.txt
+			Find wird in die datei ausgaben.txt umgeleitet.
+	
+	Standarfehlerausgabe:
+		Zweiter Ausgabestrom, der für Fehler- und Statusmeldungen gedacht ist.
+			Standartfehlerausgaben haben die Gerätedatei /dev/stderr, Deskriptor Num. 2
+
+	Pipe:
+		Eine Pipe / Pipeline ist ein Datenstrom zwischen zwei Prozessen.
+		Funktioniert nach dem first in first out Prinzip.
+		Das ergebnis eines Programms wird dabei als Eingabe eines anderen Programms verwendet.
+		In der shell wird dies durch den "|" realisiert. So kann man mehrere befehle hintereinander Ausführen.
```

Comment:
Clarification needed:

1. Can a pipe be used for more or less than two processes?